opencart 3.0.0.0 - 3.0.3.1/upload/admin/controller/extension/module/smaily_for_opencart.php

```diff
@@ -392,21 +365,38 @@ public function ajaxValidateCredentials() {
             $response = [];
             // Language for resonse.
             $this->load->language('extension/module/smaily_for_opencart');
+            
             // Check if all fields are set.
-            if( !empty($this->request->post['subdomain'] &&
-                !empty($this->request->post['username']) &&
-                !empty($this->request->post['password']))) {
-                    $subdomain = $this->db->escape($this->request->post['subdomain']);
-                    $username =  $this->db->escape($this->request->post['username']);
-                    $password = $this->db->escape($this->request->post['password']);
-                    // Validate credentials with a call to Smaily.
-                    $validate = $this->validateSmailyCredentials($subdomain, $username, $password);
-                } else {
+            if (empty($this->request->post['subdomain']) ||
+                empty($this->request->post['username']) ||
+                empty($this->request->post['password'])) {
                     // Show empty field error message.
                     $response['error'] = $this->language->get('error_validate_empty');
-                }
+                    echo json_encode($response);
+                    return;
+            }
+
+            $subdomain = $this->db->escape($this->request->post['subdomain']);
```

Comment:
Why are you escaping the value for database insert before cleaning it?

Reply:
Moved subdomain escaping after cleaning https://github.com/sendsmaily/smaily-opencart-module/pull/1/commits/5bedbd12ccd5810dcc735616738eab07290ce006